mpi4jax/__init__.py

```diff
@@ -23,6 +23,7 @@
 
 __all__ = [
     "Allreduce",
+    "AllreduceT",
```

Comment:
surely we don't need to export such implementation detail. also above

Reply:
I think we need to export it, since we want our users to be able to also define the transposed version of matrix-vector multiplication which transposes correctly, see `matvec_transpose_mpi` in https://github.com/PhilipVinc/mpi4jax/issues/34#issuecomment-742454368

(in netket we would need it here: https://github.com/netket/netket/blob/fabb3c55fa45387cbe2080359e4cf6901ac80e45/netket/optimizer/jax/stochastic_reconfiguration.py#L21)

Reply:
ok, you convinced me.